crates/pessimistic-proof/src/local_state.rs

```diff
@@ -0,0 +1,62 @@
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    certificate::Certificate,
+    local_balance_tree::BalanceTree,
+    local_exit_tree::{hasher::Keccak256Hasher, LocalExitTree},
+    proof::{BalanceRoot, ExitRoot},
+    ProofError,
+};
+
+/// Local state of one network.
+/// The AggLayer tracks the [`LocalNetworkState`] for all networks.
+/// Eventually, this state will be entirely tracked by the networks themselves.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct LocalNetworkState {
+    /// Commitment to the [`BridgeExit`].
+    pub exit_tree: LocalExitTree<Keccak256Hasher>,
+    /// Commitment to the balance for each token.
+    pub balance_tree: BalanceTree,
+}
+
+impl LocalNetworkState {
+    /// Apply the [`Certificate`] on the current [`State`].
+    /// Returns the commitment on the resulting state if successful.
+    pub fn apply_certificate(
+        &mut self,
+        certificate: Certificate,
```

Comment:
I think that `Certificate` can be passed as ref

Reply:
done